lib/member.symphony.php

```diff
@@ -372,6 +372,80 @@ public function filter_LockActivation(array &$context) {
 		}
 
 		/**
+		 * Validate Password
+		 * If there is an authentication field, we can check if a posted password is
+		 * valid for the currently logged-in Member. This filter is supposed to be
+		 * called as a pre-save filter, so it can terminate a section event (thus
+		 * prevent saving) if no valid password is provided.
+		 *
+		 * @param array $context
+		 */
+		public function filter_ValidatePassword(array &$context) {
+			$member = $this->Member;
+			if (!$member instanceof Entry) {
+				$context['messages'][] = array(
+					'member-validate-password',
+					false,
+					__('Member not found.'),
+					array(
+						'message-id' => MemberEventMessages::MEMBER_INVALID
+					)
+				);
+
+				return;
+			}
+
+			if (is_null($this->section->getFieldHandle('authentication'))) {
+				$context['messages'][] = array(
+					'member-validate-password',
+					false,
+					__('No Authentication field found.'),
+					array(
+						'message-id' => MemberEventMessages::MEMBER_ERRORS
+					)
+				);
+
+				return;
+			}
+
+			$member_id = null;
+			if (isset($_POST['fields'][$this->section->getFieldHandle('authentication')]['validate'])) {
+				$password = Symphony::Database()->cleanValue(
```

Comment:
We do not need to clean the password, as it will be hashed anyways.

Reply:
Yes, you are right.

Reply:
See 3216572